src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -539,7 +539,7 @@ public Long dbSize() {
    *         2.1.3, Redis &gt;= 2.1.3 will happily update the timeout), or the key does not exist.
    */
   @Override
-  public Long expire(final byte[] key, final int seconds) {
+  public Long expire(final byte[] key, final long seconds) {
```

Comment:
This is already merged now, but [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) might be a better unit for the timeout.

Reply:
@walles I'm not sure if you noticed, this not related to socket timeout. This is actually [EXPIRE](https://redis.io/commands/expire) command.

Still, this can be considered. But let's rebuild Jedis first :)